### DISCUSSION TITLE
`moduleResolution:nodenext` causes `tailwindcss()`  not callable

### DISCUSSION THREAD

**Discussion by Junkher** [tailwindlabs/tailwindcss#9470] [Help]
**What version of Tailwind CSS are you using?**

v3.1.8

**What build tool (or framework if it abstracts the build tool) are you using?**

vite

**What version of Node.js are you using?**

v16.15.1

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows 10

**Reproduction URL**

[Playground link with relevant code](https://www.typescriptlang.org/play?target=99&moduleResolution=99&module=199&ts=4.7.4&jsx=1#code/JYWwDg9gTgLgBDAhsANgd2AOwCYGMDO+cAZlBCHAERKoY4H6UDcAUKJLHIgK4wRhQApsWAAPQVBJkKlHnwHCxE5ixY10WPIQAUASlZz+QkeKh6mQA)

The playground doesn’t save the config. I can force the error to appear by setting the module configuration in the settings.

![image](https://user-images.githubusercontent.com/85776620/193500414-287eb3cd-e47f-4076-9af4-bf8bc31dae59.png)

**Describe your issue**

`moduleResolution:NodeNext` causes errors `This expression is not callable.`

**Comment by RobinMalfait** [comment]
Hey! Thank you for your question!
Much appreciated! 🙏

You typically don't import `tailwindcss` itself in your TypeScript application. Tailwind is used in your build chain as a PostCSS plugin, so there shouldn't be any TypeScript issues when integrating Tailwind with PostCSS.

Going to move this to a discussion, since this doesn't seem like a bug at first sight to me. But I'm curious to know what your use case is exactly.

**Reply by Junkher** [reply]
Actually I used it as a PostCss plugin in my vuepress-next theme. The relevant code is as below:
```
  app.options.bundler = viteBundler({
          viteOptions: {
            css: {
              postcss: {
                plugins: [
                  tailwindcss(),
                  autoprefixer(),
                ],
              },
            },
          },
        })
```
The `tailwindcss()` causes such an error, while `autoprefixer()` works well.

**Comment by Alxandr** [comment]
I just ran into issues trying to create a tailwind plugin in typescript.

**Comment by tycrek** [comment]
I'm running into this as well. Any updates?
